f8a_worker/storages/postgres.py

```diff
@@ -278,3 +278,17 @@ def get_analysed_versions(ecosystem, package):
         except SQLAlchemyError:
             PostgresBase.session.rollback()
             raise
+
+
+def validate_utf_json(result):
+    """Remove non UTF-8 characters from description if present.
+
+    :param result: dict, Dictionary having details of task
+    return: a dictionary received as input having non UTF characters removed if present
+    """
+    if 'details' in result and isinstance(result['details'], list) \
+            and len(result['details']) > 0 and 'description' in result['details'][0]:
+        result['details'][0]['description'] = ''.join(
+            x for x in result['details'][0]['description']
```

Comment:
i dont understand this step. there are ways in python to convert unicode to utf8
1. your_str.decode('utf-8', 'ignore') or
2. unicode(your_str, "utf-8")

Reply:
Yes, other functions are also there but seems this approach gives us better output as it keeps only printable characters.
unicode() function was there in Python 2 but it was replaced in Python 3 with str.
Please have a look on output using different functions:

#Input string
test_str = '��C\x00h\x00a\x00n\x00g\x00e\x00 \x001\x00'

1) ''.join(x for x in s if x in string.printable)
#Output = 'Change 1'

2) test_str.encode('utf-8','ignore').decode('utf-8','ignore')
#Output = '��C\x00h\x00a\x00n\x00g\x00e\x00 \x001\x00'

3) str(test_str.encode('utf-8'))
#Output = "b'\\xef\\xbf\\xbd\\xef\\xbf\\xbdC\\x00h\\x00a\\x00n\\x00g\\x00e\\x00 \\x001\\x00'"

Reply:
why are you doing encode and decode in option number 2?

Reply:
It is because str.encode() gives us Bytes, so we need convert it into String, for that we can use decode() as given in option 2 or str() as given in option 3.